docs/proposals/pod-resource-management.md

```diff
@@ -401,6 +402,39 @@ Update Kubelet’s metrics provider to include Pod level metrics. Use cAdvisor's
 
 This feature will be opt-in in v1.4 and an opt-out in v1.5. We recommend users to drain their nodes and opt-in, before switching to v1.5, which will result in a no-op when v1.5 kubelet is rolled out.
 
+## Implementation Status
+
+The implementation goals of the first milestone are outlined below.
+- [x] Finalize and submit Pod Resource Management proposal for the project #26751
+- [x] Refactor qos package to be used globally throughout the codebase #27749 #28093
+- [x] Add interfaces for CgroupManager and CgroupManagerImpl which implements the CgroupManager interface and creates, destroys/updates cgroups using the libcontainer cgroupfs driver. #27755 #28566
+- [x] Inject top level QoS Cgroup creation in the Kubelet and add e2e tests to test that behaviour. #27853
+- [x] Add PodContainerManagerImpl Create and Destroy methods which implements the respective PodContainerManager methods using a cgroupfs driver. #28017
+- [x] Have docker manager create container cgroups under pod level cgroups. Inject creation and deletion of pod cgroups into the pod workers. Add e2e tests to test this behaviour. #29049
+- [x] Add support for updating policy for the pod and QoS containers. Add e2e tests to test this behaviour. #29087
+- [ ] Figure out a method to restart containers having Restart Policy: Never, which are killed intermittently when they are first started.
+- [ ] Manage QoS level cgroup parameter updates when a pod is added or deleted from the kubelet. Can use cadvisor's watch method to track updates/changes on the cgroup mount One option is to do this through a reconcilation loop that ensures pod cgroups on the cgroup mount are in the desired state. ie. only the desired cgroups exist each with the desired resource parameters. This loop would also ensure that the top level cgroups exist, So it will recreate the top level cgroups if they are deleted by some rogue process on the system.
+- [ ] [TEST] Roll backs and roll forward should be accompanied by detailed error message when killing pod intermittently during Rollout and Rollback
```

Comment:
Sorry for the review delay.  This is on my queue for tomorrow!

On Friday, July 29, 2016, Buddha Prakash <notifications@github.com> wrote:

> In docs/proposals/pod-resource-management.md
> <https://github.com/kubernetes/kubernetes/pull/29583#discussion_r72873072>
> :
>
> > @@ -401,6 +402,39 @@ Update Kubelet’s metrics provider to include Pod level metrics. Use cAdvisor's
> >
> >  This feature will be opt-in in v1.4 and an opt-out in v1.5. We recommend users to drain their nodes and opt-in, before switching to v1.5, which will result in a no-op when v1.5 kubelet is rolled out.
> >
> > +## Implementation Status
> > +
> > +The implementation goals of the first milestone are outlined below.
> > +- [x] Finalize and submit Pod Resource Management proposal for the project #26751
> > +- [x] Refactor qos package to be used globally throughout the codebase #27749 #28093
> > +- [x] Add interfaces for CgroupManager and CgroupManagerImpl which implements the CgroupManager interface and creates, destroys/updates cgroups using the libcontainer cgroupfs driver. #27755 #28566
> > +- [x] Inject top level QoS Cgroup creation in the Kubelet and add e2e tests to test that behaviour. #27853
> > +- [x] Add PodContainerManagerImpl Create and Destroy methods which implements the respective PodContainerManager methods using a cgroupfs driver. #28017
> > +- [x] Have docker manager create container cgroups under pod level cgroups. Inject creation and deletion of pod cgroups into the pod workers. Add e2e tests to test this behaviour. #29049
> > +- [x] Add support for updating policy for the pod and QoS containers. Add e2e tests to test this behaviour. #29087
> > +- [ ] Figure out a method to restart containers having Restart Policy: Never, which are killed intermittently when they are first started.
> > +- [ ] Manage QoS level cgroup parameter updates when a pod is added or deleted from the kubelet. Can use cadvisor's watch method to track updates/changes on the cgroup mount One option is to do this through a reconcilation loop that ensures pod cgroups on the cgroup mount are in the desired state. ie. only the desired cgroups exist each with the desired resource parameters. This loop would also ensure that the top level cgroups exist, So it will recreate the top level cgroups if they are deleted by some rogue process on the system.
> > +- [ ] [TEST] Roll backs and roll forward should be accompanied by detailed error message when killing pod intermittently during Rollout and Rollback
>
> Yes. Will make it more specific.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29583/files/4ca4d8c008284c5a57d8c2417f93662cf05cc45e#r72873072>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF8dbGEMDo_xCBkVzJIUGzpU1OjJ9B8Eks5qaogugaJpZM4JUp1l>
> .
>